scan-engine/src/main/java/de/tillhub/scanengine/generic/GenericKeyEventScanner.kt

```diff
@@ -0,0 +1,58 @@
+package de.tillhub.scanengine.generic
+
+import android.app.Activity
+import android.view.KeyEvent
+import android.view.View
+import android.widget.EditText
+import de.tillhub.scanengine.KeyEventScanner
+import de.tillhub.scanengine.barcode.BarcodeScannerImpl
+import de.tillhub.scanengine.data.Scanner
+import de.tillhub.scanengine.data.ScannerEvent
+import de.tillhub.scanengine.data.ScannerResponse
+import de.tillhub.scanengine.data.ScannerType
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.emptyFlow
+
+internal class GenericKeyEventScanner(
+    private val activity: Activity,
+    mutableScannerEvents: MutableStateFlow<ScannerEvent>,
+) : BarcodeScannerImpl(mutableScannerEvents), KeyEventScanner {
+
+    private val inputBuffer = StringBuilder()
+    private var lastEventTime: Long = 0L
+    private val scanThreshold = 50L
+
+    override fun startPairingScreen(scanner: ScannerType) = Unit
+    override fun observeScanners(): Flow<List<Scanner>> = emptyFlow()
+    override suspend fun disconnect(scannerId: String): ScannerResponse = ScannerResponse.Success.Disconnect
+    override suspend fun connect(scannerId: String): ScannerResponse = ScannerResponse.Success.Connect
+
+    override fun dispatchKeyEvent(event: KeyEvent, scanKey: String?) {
```

Comment:
Write here small comment how this works and how do we distinguish between keyboards and scanners.